src/metrics/timingMetric.js

```diff
@@ -0,0 +1,43 @@
+import {VALID_TYPES} from './metrics';
+
+function typeOrError(type) {
+  if (VALID_TYPES.includes(type)) {
+    return type;
+  }
+  throw new Error('invalid type for a metric supplied');
+}
+
+export default function newMetric(metricType, {fields, tags} = {
+  fields: [],
+  tags: []
+}, stop = undefined, start = undefined) {
```

Comment:
Just curious, but is there any reason with this setup that stop would be manually supplied instead of called as a function? To me it just feels like potential abuse cases in an API. I somewhat get why you might want to allow `start`, but stop just feels weird here.